src/box/memtx_engine.cc

```diff
@@ -1552,6 +1552,12 @@ memtx_engine_memory_stat(struct engine *engine, struct engine_memory_stat *stat)
 							MEMTX_EXTENT_SIZE;
 }
 
+static void
+memtx_engine_switch_to_ro(struct engine *engine)
+{
+	memtx_tx_abort_writers_for_ro(engine);
+}
```

Comment:
Will applier txns be aborted, when e.g. the node switches from multimaster to single master? They should not and AFAICS, they'll now

And it'd be great to add that to tests also

Reply:
They won't, appliers use `n_applier_rows` instead of `n_new_rows`, as stated by @LevKats

Reply:
SInce the first yield will be when txn is already in prepared state and cannot be aborted (during commit), this cannot be reproduced, applier transactions won't be aborted, test is not needed

Reply:
There's a yield during checking format in `on_replace` for `_space`. We can try to abuse that and make the test independent from the engine: create two masters, set `ERRINJ_CHECK_FORMAT_DELAY_COUNTDOWN` to 1 on the first one, on the second one - `space:alter`, wait on the first one until `ERRINJ_CHECK_FORMAT_DELAY` is `true`, call` box.cfg{read_only = true}` there and make sure, that the txn is applied.